src/cf/client.go

```diff
@@ -186,6 +199,21 @@ func (c *cfClient) GetTokens() Tokens {
 	return c.tokens
 }
 
+func (c *cfClient) GetTokensWithRefresh() Tokens {
+	if c.isTokenToBeExpired() {
```

Comment:
yes there is potential data race for  expire time. I will remove the outer check.